rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/LightModelClientWrapper.kt

```diff
@@ -101,4 +115,54 @@ class LightModelClientWrapper(
         return this.lightModelClient.Area()
     }
 
+
+
+    suspend fun updateRoom(newRoom: Room){
+        var result: N_Room? = null
+        val root = lightModelClient.waitForRootNode()
+        if (root != null) {
+
+            var decodedReference = URLDecoder.decode(newRoom.roomRef, Charset.defaultCharset())
+            val result = resolveNodeIdToConcept(decodedReference) as N_Room
+            // test these
+            // "the right way"
+            //INodeReferenceSerializer.deserialize(actualRef).resolveNode(lightModelClient.getRootNode()?.getArea())
+
+            // warning not performant!
+            // root.typed<N_Repository>().descendants(true).ofType<N_Room>()
+
+            lightModelClient.runWrite {
+                result.name = newRoom.name
+                result.maxPlaces = newRoom.maxPlaces
+                result.hasRemoteEquipment = newRoom.hasRemoteEquipment!!
+            }
+        }
```

Comment:
```suggestion
        val root = lightModelClient.waitForRootNode() ?: return
        
        val decodedReference = URLDecoder.decode(newRoom.roomRef, Charset.defaultCharset())
        val result = resolveNodeIdToConcept(decodedReference) as N_Room
        // test these
        // "the right way"
        //INodeReferenceSerializer.deserialize(actualRef).resolveNode(lightModelClient.getRootNode()?.getArea())

        // warning not performant!
        // root.typed<N_Repository>().descendants(true).ofType<N_Room>()

        lightModelClient.runWrite {
            result.name = newRoom.name
            result.maxPlaces = newRoom.maxPlaces
            result.hasRemoteEquipment = newRoom.hasRemoteEquipment!! 
        }
```

if can be shortened with ?: and decodedReference can be val